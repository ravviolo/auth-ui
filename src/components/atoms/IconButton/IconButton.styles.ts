import styled from 'styled-components';

export const IconButton = styled.button`
  background-color: transparent;
  border: 2px solid;
  height: 2rem;
  width: 2rem;
  border-radius: 100%;
  border-width: 2px;
  font-weight: bold;
  font-size: 1.2rem;
  text-align: center;
  cursor: pointer;
`;
