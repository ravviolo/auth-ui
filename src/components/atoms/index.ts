export { Button } from './Button/Button';
export { Checkbox } from './Checkbox/Checkbox';
export { Divider } from './Divider/Divider';
export { Heading } from './Heading/Heading';
export { IconButton } from './IconButton/IconButton';
export { InputField } from './InputField/InputField';
export { InputLabel } from './InputLabel/InputLabel';
export { TextButton } from './TextButton/TextButton';
