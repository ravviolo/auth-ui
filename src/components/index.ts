export * from './templates/LoginTemplate/LoginTemplate';
export * from './templates/SignUpTemplate/SignUpTemplate';
