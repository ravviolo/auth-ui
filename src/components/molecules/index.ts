export { AuthFooter } from './AuthFooter/AuthFooter';
export { AuthHeader } from './AuthHeader/AuthHeader';
export { AuthSocials } from './AuthSocials/AuthSocials';
export { EmailInput } from './EmailInput/EmailInput';
export { PasswordInput } from './PasswordInput/PasswordInput';
