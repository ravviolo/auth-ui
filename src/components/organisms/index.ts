export * from './LoginForm/LoginForm';
export * from './SignUpForm/SignUpForm';
