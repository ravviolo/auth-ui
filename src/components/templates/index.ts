export * from './LoginTemplate/LoginTemplate';
export * from './SignUpTemplate/SignUpTemplate';
