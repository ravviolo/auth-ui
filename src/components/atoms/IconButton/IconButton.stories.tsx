import { ComponentMeta, ComponentStory } from '@storybook/react';

import { IconButton } from 'components/atoms';

export default {
  title: 'Atoms/IconButton',
  component: IconButton,
  args: {
    icon: 'F',
    type: 'button',
  },
} as ComponentMeta<typeof IconButton>;

export const Default: ComponentStory<typeof IconButton> = (args) => (
  <IconButton {...args} />
);
Default.storyName = 'IconButton';
