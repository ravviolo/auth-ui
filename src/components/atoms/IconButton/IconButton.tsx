import * as S from './IconButton.styles';

export interface Props {
  icon: string;
  type: 'button' | 'submit' | 'reset';
  color: string;
  testId: string;
  className?: string;
  onClick?: () => void;
}

export const IconButton = ({
  icon,
  color,
  testId,
  className,
  type = 'button',
  onClick,
}: Props) => {
  return (
    <S.IconButton
      className={className}
      color={color}
      data-testid={testId}
      type={type}
      onClick={onClick}
    >
      {icon}
    </S.IconButton>
  );
};
