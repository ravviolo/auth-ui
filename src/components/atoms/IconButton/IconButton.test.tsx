import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { create } from 'react-test-renderer';

import { IconButton } from 'components/atoms';

import { Props as IconButtonProps } from './IconButton';

describe('IconButton', () => {
  const mockOnClick = jest.fn<IconButtonProps['onClick'], []>();

  const props: IconButtonProps = {
    color: 'cyan',
    icon: 'F',
    type: 'button',
    testId: 'icon-btn-test-id',
    onClick: mockOnClick,
  };

  it('should render button with specified color and icon', () => {
    render(<IconButton {...props} />);

    const iconBtn = screen.getByTestId('icon-btn-test-id');

    expect(iconBtn).toHaveTextContent('F');

    const iconBtnStyle = window.getComputedStyle(iconBtn);

    expect(iconBtnStyle.color).toBe('cyan');
    expect(iconBtnStyle.borderColor).toBe('cyan');
  });

  it('should handle click event, fires event handler once', async () => {
    render(<IconButton {...props} />);

    const iconBtn = screen.getByTestId('icon-btn-test-id');

    await userEvent.click(iconBtn);

    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });

  it('should match snapshot', () => {
    const tree = create(<IconButton {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
