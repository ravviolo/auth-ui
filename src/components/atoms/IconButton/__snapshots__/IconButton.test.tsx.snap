// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should match snapshot 1`] = `
<button
  className="sc-iBkjds fegfkl"
  color="cyan"
  data-testid="icon-btn-test-id"
  onClick={[MockFunction]}
  type="button"
>
  F
</button>
`;
